src/util/db.js

```diff
@@ -0,0 +1,155 @@
+      params.Key = { id: item }
+      return await db.get(params).promise()
+        .then(({ Item }) => Item)
+        .catch(e => console.log(e))
+    } else {
+      return await db.scan(params).promise()
+        .then(({ Items }) => Items)
+        .catch(e => console.log(e))
+    }
+  },
+
+  post: async (item) => {
+    const params = {
+      TableName: 'jobs',
+      Item: item
+    }
+    return await db.put(params).promise()
+  },
+
+  update: async (jobId, job) => {
+    const params = {
+      TableName: 'jobs',
+      Key: {
+        id: jobId
+      },
+      UpdateExpression: 'set attendance = :attendance, categories = :categories, description = :description, #duration = :duration, engagement = :engagement, experience = :experience, location_req = :location_req, start_date = :start_date, title = :title, weekly_hours = :weekly_hours, rate_client = :rate_client, rate_esteemed = :rate_esteemed, #timezone = :timezone, skills = :skills',
+      ExpressionAttributeValues: {
+        ':attendance': job.attendance,
+        ':categories': job.categories,
+        ':description': job.description,
+        ':duration': job.duration,
+        ':engagement': job.engagement,
+        ':experience': job.experience,
+        ':location_req': job.location_req,
+        ':start_date': job.start_date,
+        ':title': job.title,
+        ':weekly_hours': job.weekly_hours,
+        ':rate_client': job.rate_client,
+        ':rate_esteemed': job.rate_esteemed,
+        ':timezone': job.timezone,
+        ':skills': job.skills
+      },
+      ExpressionAttributeNames: {
+        '#duration': 'duration',
+        '#timezone': 'timezone'
+      }
+    }
+
+    if (job.active !== undefined) {
+      params.UpdateExpression = params.UpdateExpression + ', active = :active'
+      params.ExpressionAttributeValues[':active'] = job.active
+    }
+
+    db.update(params).promise()
+  },
+
+  updateNotes: async (jobId, notes) => {
+    const params = {
+      TableName: 'jobs',
+      Key: {
+        id: jobId
+      },
+      UpdateExpression: 'set notes = :notes',
+      ExpressionAttributeValues: {
+        ':notes': notes
+      }
+    }
+    return await db.update(params).promise()
+  },
+
+  saveApplication: async (jobId, applicants) => {
+    const params = {
+      TableName: 'jobs',
+      Key: {
+        id: jobId
+      },
+      UpdateExpression: 'set applicants = :applicants',
+      ExpressionAttributeValues: {
+        ':applicants': applicants
+      }
+    }
+
+    return await db.update(params).promise()
+  }
+}
+
+// Profile Calls
+const profiles = {
+  get: async (userId) => {
+    const params = {
+      TableName: 'profiles',
+      Key: {
+        id: userId
+      }
+    }
+    return (await db.get(params).promise().then(({ Item }) => Item) || {})
+  },
+
+  getAll: async () => {
+    const params = {
+      TableName: 'profiles'
+    }
+    return await db.scan(params).promise().then(({ Items }) => Items)
+  },
+
```

Comment:
This function now needs an object as the 2nd argument. The updateExpression item will be constructed by the key and value of that object.

The third argument needs to be an array of the items to be updated, and will associated with the value of that 2nd argument.

I hope that makes sense.

Reply:
So, why do we need a 3rd argument? That was my intention with the object. `{ field: value, ... }`

Reply:
The array of fields that need to be updated would be the keys from that 2nd argument.

Reply:
I'm afraid I don't understand. The values to be changed on a few occasions are items nested in other items, and while the path does contain the key from the 2nd argument, there's more to it. 
For instance : 'values.wp_experience.val.selected_option.value' in wp.js

Reply:
So, in that instance, the object passed in would be-
```
{ wp_experience: values.wp_experience.val.selected_option.value }
```

Reply:
Got it. In that case then, there will need to be more than just the one update method since the tasks update would not work in that instance since the Update Expression is not two like terms 

( UpdateExpression: 'set tasks = :v', ) 
vs 
(UpdateExpression: 'set drupal_profile = :drupal_profile, drupal_bio = :drupal_bio',)